src/main/java/de/tum/cit/aet/artemis/core/service/feature/FeatureToggleService.java

```diff
@@ -73,23 +93,34 @@ public void enableFeature(Feature feature) {
      * @param feature The feature that should be disabled
      */
     public void disableFeature(Feature feature) {
-        features.put(feature, false);
-        sendUpdate();
+        getFeatures().ifPresent(features -> {
+            features.put(feature, false);
+            sendUpdate();
+        });
     }
 
     /**
      * Updates the given feature toggles and enables/disables the features based on the given map. Also notifies all clients
      * by sending a message via the websocket.
      *
-     * @param features A map of features (feature -> shouldBeActivated)
+     * @param updatedFeatures A map of features (feature -> shouldBeActivated)
      */
-    public void updateFeatureToggles(final Map<Feature, Boolean> features) {
-        this.features.putAll(features);
-        sendUpdate();
+    public void updateFeatureToggles(final Map<Feature, Boolean> updatedFeatures) {
+        getFeatures().ifPresent(features -> {
+            features.putAll(updatedFeatures);
+            sendUpdate();
+        });
     }
 
     private void sendUpdate() {
-        websocketMessagingService.sendMessage(TOPIC_FEATURE_TOGGLES, enabledFeatures());
+        try {
+            if (hazelcastInstance != null && hazelcastInstance.getLifecycleService().isRunning()) {
```

Comment:
We have this check `hazelcastInstance != null && hazelcastInstance.getLifecycleService().isRunning()` repeated quite some time, could we move it to a method instead?

This would make adjustments in the future much easier in case another edge case is found